docs/guides/workflows_patterns/deployment_airflow.rst

```diff
@@ -1,17 +1,106 @@
+    - Optional: Configured a :ref:`Checkpoint <tutorials__getting_started__set_up_your_first_checkpoint>`.
+    - Configured an Airflow pipeline (DAG).
+
+There are two supported methods: using an Airflow ``PythonOperator`` to run Validations using python code or invoking the Great Expectations CLI to run a Checkpoint using an Airflow ``BashOperator``.
+
+.. note::
+    There is not currently a Great Expectations custom Airflow operator, however we have heard from users who have implemented their own. We love community contributions so please share your operators!
+
+Check out these links for some examples of Airflow pipelines with Great Expectations Validations ensuring that downstream tasks are protected from upstream issues:
+
+- `Great Expectations Pipeline Tutorial <https://github.com/superconductive/ge_tutorials>`_
+- `Great Expectations / Airflow Demo <https://github.com/superconductive/airflow_meetup_demo>`_
+
+In the first link and the diagram below, you can see a common pattern of using Validation tasks to ensure that the data flowing from one task to the next is correct, and alert the team if it is not. Another common pattern is to branch and change your DAG based on a Validation (e.g. send data for more cleaning before moving to the next task, store it for a postmortem, etc.).
+
+.. image:: ge_tutorials_pipeline.png
+    :width: 800
+    :alt: Airflow pipeline from Great Expectations tutorials repository.
+
+``PythonOperator``
+------------------
+
+1. **Create Validation Methods**
+
+    Create the methods to validate data that will be called in your DAG. In this example our data is contained in a file.
+
+.. code-block:: python
+
+    from airflow import AirflowException
+    from airflow.operators.python_operator import PythonOperator
+    import great_expectations as ge
+
+    ...
+
+    def validate_data(ds, **kwargs):
+
+        # Retrieve your data context
+        context = ge.data_context.DataContext(<insert path to your great_expectations.yml>)
+
+        # Create your batch_kwargs
+        batch_kwargs_file = {
+            "path": <insert path to your data file>,
+            "datasource": "input_files"}
+
+        # Create your batch (batch_kwargs + expectation suite)
+        batch_file = context.get_batch(batch_kwargs_file, <insert name of your expectation suite>)
+
+        # Run the validation
+        results = context.run_validation_operator(
+            "action_list_operator",
+            assets_to_validate=[batch_file],
+            # This run_id can be whatever you choose
+            run_id=f"airflow: {kwargs['dag_run'].run_id}:{kwargs['dag_run'].start_date}")
+
+        # Handle result of validation
+        if not results["success"]:
+            raise AirflowException("Validation of the data is not successful ")
+
+
+2. **Add Validation Methods to DAG**
+
+    Validation steps can be added after pipeline steps to ensure that the steps were completed successfully.
```

Comment:
"pipeline steps" doesn't quite sound right (since GE would technically also be part of the pipeline... whatever the definition of pipeline is ;) , maybe something like data transformation or loading steps?

Reply:
Agreed - I changed to `data retrieval, transformation or loading steps` to include retrieval (since it's in the example, and in our new case study :)